hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -98,6 +99,11 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         }
     }
 
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        // nothing to do
```

Comment:
Here we should check if the `contractID` is set.

Reply:
In preHandle (line 88) we call `op.contractIDOrElse(ContractID.DEFAULT)`. Is it correct to use a default contract ID if no contract ID is set?